src/main/java/domain/Lotto.java

```diff
@@ -1,21 +1,26 @@
 package domain;
 
-import java.util.Collections;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 public class Lotto {
+    public static final int LOTTO_PRICE = 1000;
     private static final int LOTTO_SIZE = 6;
     private static final String ERROR_MESSAGE_FOR_INVALID_SIZE_OF_LOTTO_NUMBERS = "%d개의 숫자를 골라주세요.";
     private static final String ERROR_MESSAGE_FOR_DUPLICATE_LOTTO_NUMBERS = "숫자는 중복될 수 없습니다.";
+    private static final String ERROR_MESSAGE_FOR_NULL_LOTTO_NUMBERS = "로또 번호가 비었습니다";
 
     private final List<LottoNumber> lottoNumbers;
 
-    public Lotto(final List<LottoNumber> lottoNumbers) {
-        final List<LottoNumber> unmodifiableLottoNumbers = Collections.unmodifiableList(lottoNumbers);
-        validateSize(unmodifiableLottoNumbers);
-        validateDuplicate(unmodifiableLottoNumbers);
+    public Lotto(List<LottoNumber> lottoNumbers) {
+        Objects.requireNonNull(lottoNumbers, ERROR_MESSAGE_FOR_NULL_LOTTO_NUMBERS);
 
-        this.lottoNumbers = unmodifiableLottoNumbers;
+        List<LottoNumber> copiedLottoNumbers = List.copyOf(lottoNumbers);
+        validateSize(copiedLottoNumbers);
+        validateDuplicate(copiedLottoNumbers);
+
+        this.lottoNumbers = copiedLottoNumbers;
```

Comment:
```suggestion
        validateSize(lottoNumbers);
        validateDuplicate(lottoNumbers);
        this.lottoNumbers  = List.copyOf(lottoNumbers);
```

파라미터로 받은 객체를 validate 하고 마지막에 초기화해도 괜찮지 않을까요~?

Reply:
저도 그게 좋은 것 같아요! 

사실 이 부분 저도 그렇게 하고 싶었는데..
생성자 내에서 검증 메서드 선언 순서가 바뀌는 경우에 대해 생각해보라는
피드백을 본 적이 있어서.. 받자마자 컬렉션 수정을 막아야되겠다고 생각했었어요.

제시해주신 코드로 반영했습니다 감사합니다 루피! 😃